Выполнение урока 15

carrydan opened the pull request:
Привет. Выполнил урок 15.

carrydan committed:
выполнение урока 15

raguzin reviewed (changes requested):
реализация вероятно рабочая, но:
1. если мы имеем два модуля, то нужно это использовать. Для чего в модуле создания ВМ мы опять получаем инфо о подсетях, если мы это уже получили в модуле для сети.
2. вызов модулей нужно производить из "корневого" модуля, при этом при вызове модуля ВМ, ему передаем переменную с subnet_id определенным в зависимости от zone.
3. по best practice, в модуле должны присутствовать 3 файла variables, main, output И по ним разносится соответствующая конфигурация

carrydan committed:
изменил файлы.

В terraform.tfvars, копируем subnet_id после получения подсетей, для разворачивания ВМ.

raguzin reviewed (changes requested):
Лучше, но выбор подсети должен выполняться не вручную, через передачу переменной из tfvars
```subnet_id = var.selected_subnet_id```
а автоматически на основе переданной zone.
Примерно так
```subnet_id = module.data_subnet.subnets_info[var.default_zone].id```
в зависимости от того, что там у тебя в subnets_info возвращается

carrydan committed:
Поправил

carrydan reviewed (commented):
@raguzin изменил

raguzin reviewed (approved):
ок, можно было обойтись без использования locals, выполняя вычисления сразу там, где потом подставляешь переменные localc
